rocAL/rocAL/source/readers/image/external_source_reader.cpp

```diff
@@ -0,0 +1,264 @@
+{
+    auto ret = Reader::Status::OK;
+    _folder_path = desc.path();
+    _file_id = 0;
+    _shard_id = desc.get_shard_id();
+    _shard_count = desc.get_shard_count();
+    _batch_count = desc.get_batch_size();
+    _shuffle = desc.shuffle();
+    _loop = desc.loop();
+    _mode = desc.mode();
+    _end_of_sequence = false;
+    _file_data.reserve(_batch_count);
+    return ret;
+}
+
+void ExternalSourceReader::incremenet_read_ptr()
+{
+    _read_counter++;
+    // _curr_file_idx++;
+        _curr_file_idx = (_curr_file_idx + 1) % _batch_count;
+}
+
+size_t ExternalSourceReader::open()
+{
+    if (_mode == FileMode::FILENAME) {
+        std::string next_file_name;
+        bool ret = pop_file_name(next_file_name);   // Get next file name: blocking call, will wait till next file is received from external source
+        if (_end_of_sequence && !ret) // shobi check this
+          return 0;
+        // prefix with folder_path of exists
+        if (!_folder_path.empty())
+        //   next_file_name = _folder_path + "/" + next_file_name;
+            next_file_name = next_file_name;
+        _last_id= next_file_name;
+        filesys::path pathObj(next_file_name);
+        if(filesys::exists(pathObj) && filesys::is_regular_file(pathObj))
+        {
+          _current_fPtr = fopen(next_file_name.c_str(), "rb");// Open the file,
+          if(!_current_fPtr) // Check if it is ready for reading
+              return 0;
+
+          fseek(_current_fPtr, 0 , SEEK_END);// Take the file read pointer to the end
+
+          _current_file_size = ftell(_current_fPtr);// Check how many bytes are there between and the current read pointer position (end of the file)
+
+          if(_current_file_size == 0)
+          { // If file is empty continue
+              fclose(_current_fPtr);
+              _current_fPtr = nullptr;
+              return 0;
+          }
+          fseek(_current_fPtr, 0 , SEEK_SET);// Take the file pointer back to the start
+          _file_data[_curr_file_idx] = std::make_tuple((unsigned char *)next_file_name.data(), (size_t)_current_file_size, 0, 0, 0); // shobi check typecasting
+          incremenet_read_ptr();
+        }
+    } else {
+        std::tuple<unsigned char*, size_t, int, int, int> image;
+        bool ret = pop_file_data(image);
+        if (_end_of_sequence && !ret) // shobi check this
+        {
+            std::cerr<<"\n !!!!!!!!!!!!!!!!!! EOS || POP FAILED !!!!!!!!!!!!!!!!!!!!";
+            return 0;
+        }
+        _file_data[_curr_file_idx] = image;
+        _current_file_size = std::get<1>(image);
+    }
+
+    return _current_file_size;
+}
+
+size_t ExternalSourceReader::read_data(unsigned char* buf, size_t read_size)
+{
+    if (_mode == FileMode::FILENAME) {
+        if(!_current_fPtr)
+            return 0;
+
+        // Requested read size bigger than the file size? just read as many bytes as the file size
+        read_size = (read_size > _current_file_size) ? _current_file_size : read_size;
```

Comment:
Pleasse use std::max instead of branch statement here

Reply:
I think std::min needs to be used here